src/framework/mpas_io_units.F

```diff
@@ -38,14 +46,29 @@ module mpas_io_units
 !> the unit number
 !
 !-----------------------------------------------------------------------
-    subroutine mpas_new_unit(newUnit)!{{{
+    subroutine mpas_new_unit(newUnit, unformatted)!{{{
         integer, intent(inout) :: newUnit
+        logical, optional, intent(in) :: unformatted
 
-        integer :: i
+        integer :: i, minsearch, maxsearch, limit_type
 
         logical :: opened
 
-        do i = 1, maxUnits
+        newUnit = -1
+
+        limit_type = 1
+        if ( present(unformatted) ) then
+           if ( unformatted ) limit_type = 2
+        end if
+        if ( limit_type == 2 ) then
+           minsearch = unformatted_min
+           maxsearch = unformatted_max
+        else
+           minsearch = 1
+           maxsearch = maxUnits
+        end if
```

Comment:
The logic here involving `limit_type` seems unnecessarily complicated to me (but perhaps the idea was that this would be more easily generalized in future?). Instead, it seems simpler in my opinion to use something like the following:
```fortran
        minsearch = 1
        maxsearch = maxUnits
        if ( present(unformatted) ) then
           if ( unformatted ) then
              minsearch = unformatted_min
              maxsearch = unformatted_max
           end if
        end if
```